src/java.base/share/classes/java/util/jar/JarInputStream.java

```diff
@@ -37,6 +37,50 @@
  * an optional {@code Manifest} entry. The {@code Manifest}
  * can be used to store meta-information about the JAR file and its entries.
  *
+ *  <h2>Accessing the Manifest</h2>
+ * The {@linkplain #getManifest()} method will return the {@code Manifest} if:
+ * <ul>
+ *     <li>
+ *         The {@code Manifest} is the first Jar entry or the
+ *         {@code META-INF/} is the first jar entry and {@code Manifest}
+ *         is the second jar entry
+ *     </li>
+ * </ul>
+ *
+ * <p>
+ *     The methods {@linkplain #getNextEntry()} and {@linkplain #getNextJarEntry()}
+ *     will not return the {@code META-INF/} or {@code Manifest} entries if:
+ * </p>
+ * <ul>
+ *     <li>
+ *         The {@code META-INF/} or {@code Manifest} is the first Jar entry
+ *     </li>
+ *     <li>
+ *         The {@code META-INF/} is the first jar entry and {@code Manifest}
+ *         is the second jar entry
+ *      </li>
+ * </ul>
```

Comment:
The challenge I had with the wording is due to the fact that if "META-INF/" is the first entry in the Zip file, it will not be returned regardless of whether there is a manifest.  So open to suggestions.

Reply:
True we do care more about the manifest, but was also trying to address the difference between ZipInputStream which will always return both entries regardless of location

We also have the case such where you the 1st entry in the jar is a normal file and the 2nd entry is the manifest.,  getManifest() will return null.

I guess I could say:

 _"getManifest will return the Manifest if it is the first entry or META-INF/ is  the first entry and the Manifest is the second entry within the Jar file. For all cases, getManifest will return null."

Is the above any better?

Reply:
> It's better. Do you need to explicitly say "For all other cases"?

I thought it is worth being specific,  but happy to leave it out if you and others prefer
> 
> My original comment was more about explaining `getManifest()` and `getNextEntry()` in the same if. It's still doable.

I think the challenge is there is not an exact match to the behavior between `getManifest()` and `getNext[Jar]Entry()`.    We also have the case if the manifest is the first entry and META-INF/ is the 2nd or later entry, then  META-INF/ will be returned by `JarInputStream`.  ` ZipInputStream` will return all entries which differs from `JarInputStream` so thought it made sense to be specific but perhaps I am overthinking it.

Reply:
I could do tweak further to say:

_`getManifest()` will return the Manifest if it is the first entry or META-INF/ is the first entry and the Manifest is the second entry within the Jar file.   When the Manifest is returned by `getManifest()`,  the `getNextEntry()` and `getNextJarEntry()` methods will not return the Manifest_

Is that more what you were thinking?

Reply:
I read the current draft and it makes we wonder if it would be simpler to just replace the class description with something like this:

```
 * The JarInputStream class is used to read the contents of a JAR file from
 * an input stream. It extends ZipInputStream with support for reading an optional
 * manifest named "META-INF/MANIFEST.MF" when it is found at as the first entry in
 * the stream. If the first entry in the stream is "META-INF/" then it is skipped,
 * in which case the manifest is found if it is the second entry in the stream.
 *
 * JarInputStream defines the getNextJarEntry() method to iterate through the
 * JAR file entries in the stream. The method reads the metadata for the next JAR
 * file entry in the stream and positions the input stream to read the entry's
 * file data. The first entry found by getNextJarEntry() (or getNextEntry())
 * will follow the manifest when the manifest is the first entry in the JAR
 * file (or the second entry when the first entry is "META-INF/").
```

Reply:
Thank you for the suggestions.  I just updated the the description leveraging your suggestions.  Please let me know if this is better